client/landing/gutenboarding/components/domain-picker/index.tsx

```diff
@@ -120,6 +120,24 @@ const DomainPicker: FunctionComponent< Props > = ( {
 		if ( allSuggestions ) {
 			setRailcarId( getNewRailcarId() );
 		}
+
+		if ( freeSuggestions && recommendedSuggestion && paidSuggestions ) {
+			if (
+				currentDomain?.is_free === true &&
+				currentDomain?.domain_name !== freeSuggestions[ 0 ].domain_name
+			) {
+				onDomainSelect( freeSuggestions[ 0 ] );
+			}
+
+			if (
+				currentDomain?.is_free !== true &&
+				! paidSuggestions.find(
+					( suggestion ) => suggestion.domain_name === currentDomain?.domain_name
+				)
+			) {
+				onDomainSelect( recommendedSuggestion );
+			}
+		}
```

Comment:
There must be a better way to do this...

Reply:
I think doing it this way makes sense. We're trying to change the selection when the search results change. So I think `useEffect()` makes sense.

Maybe I would have made this a seperate `useEffect()` call? Leaving this one to just do analytics stuff, and a new one to just implement this behaviour?